reports/get_vs_marketplace_data.py

```diff
@@ -0,0 +1,157 @@
+    VSX_API = '%sapi' % API_ENDPOINT
+    
+    # Looks like 1000 is the max page size
+    get_all_extensions_payload = {
+            "assetTypes": [
+                "Microsoft.VisualStudio.Services.Icons.Default",
+                "Microsoft.VisualStudio.Services.Icons.Branding",
+                "Microsoft.VisualStudio.Services.Icons.Small"
+            ],
+            "filters": [
+                {
+                "criteria": [
+                    {
+                    "filterType": 8,
+                    "value": "Microsoft.VisualStudio.Code"
+                    },
+                    {
+                    "filterType": 10,
+                    "value": "target:\"Microsoft.VisualStudio.Code\" "
+                    },
+                    {
+                    "filterType": 12,
+                    "value": "37888"
+                    }
+                ],
+                "direction": 2,
+                "pageSize": 1000,
+                "pageNumber": 1,
+                "sortBy": 4,
+                "sortOrder": 0,
+                "pagingToken": None
+                }
+            ],
+            "flags": 870
+        }
+    all_extensions = []
+    total_all_versions = 0
+    while True:
+        response = requests.post(MS_API_URL, headers=MS_HEADERS, data=json.dumps(get_all_extensions_payload))
+        if response.status_code != 200:
+            print('HTTP %s error' % response.status_code)
+            break
+        else:
+            vsx_results = response.json()
+            extensions = vsx_results['results'][0]['extensions']
+            all_extensions = all_extensions + extensions
+            for extension in extensions:
+                versions = extension['versions']
+                total_all_versions += len(versions)
+            total = vsx_results['results'][0]['resultMetadata'][0]['metadataItems'][0]['count']
+            print('Retrieved %s of %s MS Marketplace extensions. %s total versions.' % (len(all_extensions), total, total_all_versions))
+            if len(all_extensions) == total:
+                break
+            else:
+                get_all_extensions_payload['filters'][0]['pageNumber'] = get_all_extensions_payload['filters'][0]['pageNumber'] + 1
+
+    # Output CSV File
+    csv_file = open(CSV_FILE_NAME, 'w')
+    csv_file.write("MS Publisher (Namespace), MS Extension, MS DisplayName, MS Pricing, MS Version, MS Date, VSX Version, VSX Date, VSX Publisher, VSX License, Repo\n")
+    for ext in extensions:
+        if ext['extensionName'] == '999-pse-alpine':
+            # sandy081.999-pse-alpine
+            print('here')
+        print("%s.%s" % (ext['publisher']['publisherName'], ext['extensionName']))
+        ms_extensionName, ms_publisherName, ms_displayname, ms_latest_version, ms_last_updated, ms_repo, ms_pricing = get_ms_info(ext)
+        vsx_extension_url = '%s/%s/%s?token=%s' % (VSX_API, ms_publisherName, ms_extensionName, VSX_ACCESS_TOKEN)
```

Comment:
`https://open-vsx.org/api/{namespace}/{extension}` is a public API endpoint.

Reply:
Right, I'll remove this.